File: Python/643. Maximum Average Subarray I.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        currentSum = sum(nums[:k])
        maxAvg = currentSum / k
        for i in range(k,len(nums)):
            currentSum += nums[i]
            currentSum -= nums[i-k]
            maxAvg = max(maxAvg, currentSum / k)
        return maxAvg

File: Java/643. Maximum Average Subarray I.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int currentsum = 0;
        for (int i=0;i<k;i++){
            currentsum = currentsum + nums[i];
        }
        double maxavg = (double)currentsum/k;
        for (int i=k;i<nums.length;i++){
            currentsum = currentsum + nums[i]- nums[i-k];
            maxavg = Math.max(maxavg,(double) currentsum/k);
        }
        return maxavg;
    }
}